earth/cpu_intr.c

```diff
@@ -32,13 +31,24 @@ static void timer_reset(uint core_id) {
     mtimecmp_set(mtime_get() + QUANTUM, core_id);
 }
 
+ulonglong gettime() {
+    static ulonglong last_time = 0;
+    ulonglong time = (ulonglong) mtime_get();
+    if (time < last_time) {
```

Comment:
I am confused here. Both `time` and `last_time` are `ulonglong`, which means that 0 is the lowest possible value. How could `time < last_time` ever happen then if `last_time` is initialized to 0?